.cicd/eosio-contracts-build.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+set -eo pipefail
+. ./.cicd/helpers/general.sh
+CDT_DIR_HOST=$(pwd)
+mkdir -p build_eosio_contracts 
+
+
+[[ ! -z "$CONTRACTS_VERSION" ]] || export CONTRACTS_VERSION="$(cat "$PIPELINE_CONFIG" | jq -r '.dependencies["eosio.contracts"]')"
+git clone -b "$CONTRACTS_VERSION" https://github.com/EOSIO/eosio.contracts.git 
+
+if [[ $(uname) == 'Darwin' ]]; then
+    export PATH=$CDT_DIR_HOST/build/bin:$PATH
+    cd build_eosio_contracts
+    CMAKE="cmake ../eosio.contracts"
+    echo "$ $CMAKE"
+    eval $CMAKE
+    MAKE="make -j $JOBS"
+    echo "$ $MAKE"
+    eval $MAKE
+    cd ..
+else #Linux
+    ARGS=${ARGS:-"--rm --init -v $(pwd):$MOUNTED_DIR"}
+    . $HELPERS_DIR/docker-hash.sh
+
+    PRE_CONTRACTS_COMMAND="export PATH=$MOUNTED_DIR/build/bin:$PATH && cd $MOUNTED_DIR/build_eosio_contracts"
+    BUILD_CONTRACTS_COMMAND="cmake ../eosio.contracts && make -j$JOBS"
+
+    # Docker Commands
+    # Generate Base Images
+    $CICD_DIR/generate-base-images.sh
+    if [[ "$IMAGE_TAG" == 'ubuntu-18.04' ]]; then
+        FULL_TAG='eosio/ci-contracts-builder:base-ubuntu-18.04-develop'
+        export CMAKE_FRAMEWORK_PATH="$MOUNTED_DIR/build:${CMAKE_FRAMEWORK_PATH}"
+        BUILD_CONTRACTS_COMMAND="cmake -DBUILD_TESTS=true $MOUNTED_DIR/eosio.contracts && make -j$JOBS"
+    fi
+
+    COMMANDS_EOSIO_CONTRACTS="$PRE_CONTRACTS_COMMAND && $BUILD_CONTRACTS_COMMAND"
+
+    # Load BUILDKITE Environment Variables for use in docker run
+    if [[ -f $BUILDKITE_ENV_FILE ]]; then
+        evars=""
+        while read -r var; do
+            evars="$evars --env ${var%%=*}"
+        done < "$BUILDKITE_ENV_FILE"
+    fi
+
+    DOCKER_RUN="docker run $ARGS $evars $FULL_TAG bash -c \"$COMMANDS_EOSIO_CONTRACTS\""
+    echo "$ $DOCKER_RUN"
+    eval $DOCKER_RUN
```

Comment:
Same here, as far as using the `echo`/`eval` syntax to show the customer what these two build steps are doing, if we can. This one will be harder than the last one because there is _another_ layer of BASH variable interpolation which will need _another_ layer of escape sequences....gross.

Instead of doing all that, let's just roll line 37...
```bash
COMMANDS_EOSIO_CONTRACTS="$PRE_CONTRACTS_COMMAND && $BUILD_CONTRACTS_COMMAND"
```
...into line 47:
```bash
DOCKER_RUN="docker run $ARGS $evars $FULL_TAG bash -c \"$PRE_CONTRACTS_COMMAND && $BUILD_CONTRACTS_COMMAND\""
```
This will eliminate one layer of variable interpolation, making it the same problem as the other peer review comment.

Now, we should be able to do this with line 26...
```bash
BUILD_CONTRACTS_COMMAND="CMAKE='cmake ../eosio.contracts' && echo \\\"$ \\\$CMAKE\\\" && eval \\\$CMAKE && MAKE='make -j $JOBS' && echo \\\"$ \\\$MAKE\\\" && eval \\\$MAKE"
```
...and this with line 34:
```bash
BUILD_CONTRACTS_COMMAND="CMAKE='cmake -DBUILD_TESTS=true $MOUNTED_DIR/eosio.contracts' && echo \\\"$ \\\$CMAKE\\\" && eval \\\$CMAKE && MAKE='make -j $JOBS' && echo \\\"$ \\\$MAKE\\\" && eval \\\$MAKE"
```
I think that should do it. Again, this is gross but we will fix it with contracts test scripts in story [AUTO-1375](https://blockone.atlassian.net/browse/AUTO-1375).

Reply:
Done.